examples/api/test/widgets/editable_text/text_editing_controller.0_test.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/editable_text/text_editing_controller.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Forces text to be lower case and places cursor at the end of the text field', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.TextEditingControllerExampleApp(),
+    );
+
+    const String input = 'Almost Everything Is a WIDGET! 💙';
+
+    await tester.enterText(find.byType(TextFormField), input);
+    await tester.pump();
+
+    final TextFormField textField = tester.widget(find.byType(TextFormField));
+    final TextEditingController controller = textField.controller!;
+
+    expect(find.text(input.toLowerCase()), findsOneWidget);
+    expect(controller.text, input.toLowerCase());
+    expect(
+      controller.selection,
+      const TextSelection.collapsed(offset: input.length),
```

Comment:
Isn't this default behavior? I would keep this test for the lower case functionality and write a separate test that tries to move the selection?

Reply:
@victorsanni thanks for the review!
You are right, it is the default behavior, so I added a separate test to verify that caret is positioned at the end of the input.